packages/next/src/server/async-storage/request-async-storage-wrapper.ts

```diff
@@ -147,10 +172,45 @@ export const RequestAsyncStorageWrapper: AsyncStorageWrapper<
 
         return cache.draftMode
       },
+
       reactLoadableManifest: renderOpts?.reactLoadableManifest || {},
       assetPrefix: renderOpts?.assetPrefix || '',
+      afterContext,
     }
-
     return storage.run(store, callback, store)
   },
 }
+
+function wrapCallbackForAfter<Result>(
+  renderOpts: WrapperRenderOpts | undefined,
+  callback: (store: RequestStore) => Result
+): [typeof callback, AfterContext] {
+  const isAfterEnabled = renderOpts?.experimental?.after
+  if (!(renderOpts && isAfterEnabled)) {
+    const afterContext = createDisabledAfterContext()
+    return [callback, afterContext]
+  }
+
+  const waitUntil =
+    renderOpts.waitUntil ??
+    function missingWaitUntil() {
+      // Error lazily, only when after() is actually called.
```

Comment:
Q: Why we want to error lazily here instead of telling users earlier than they're missing in impl?

Reply:
This was changed a bit, but the idea is that we want the error to be thrown at the callsite of `unstable_after` for a more sensible debugging experience. We also don't want to yell about this if the function isn't actually used. This limits the blast radius of any potential bugs in the implementation, which is useful since this is an experimental feature.